src/book.cpp

```diff
@@ -215,15 +212,42 @@ void Book::setPath(const std::string& path)
    : path;
 }
 
-const std::string& Book::getFavicon() const {
-  if (m_favicon.empty() && !m_faviconUrl.empty()) {
+const Book::Illustration Book::missingDefaultIllustration;
+
+const Book::Illustration& Book::getDefaultIllustration() const
```

Comment:
Working on the implementation of the catalog illustration endpoint (which would be base on this PR), I wonder if this method may be buggy.
The method may return a reference to the (existing) illustration but nothing guaranty the reference is keep alive.

One could call `book.getData()` (which internally call `getDefaultIllustration`) and in the same time we update the book (this is especially true when we will implement signal handling to reload the library). 
As `getDefaultIllustration` is internal and we don't keep the reference to the illustration out of Book's methods the only way to trigger this would be to have a race condition.

But in the context of kiwix-serve, we will have several thread serving the illustration and a potential reload that could appear at any time.
What do you think ? If you confirm, it is probably better to fix this now as it would be a bug pretty difficult to debug after.
Or we may define that `Library`/`Book` is not thread safe and we will have to handle this in https://github.com/kiwix/kiwix-tools/issues/243

Reply:
> One could call `book.getData()` ...

I guess you meant `book.getFavicon()`, didn't you? The issue you describe is rather a problem with the  `Book::getFavicon*()` API. So the answer depends on the what and when we are going to do with those legacy functions.

Reply:
Yes `book.getFavicon()`.
I'm in favor to drop `Book::getFavicon()`. We don't have to keep the cpp API, the next libkiwix release will break it anyway. We must use this slot to design the AP we want. Conceptually we don't have favicon now, we only and illustrations.

Reply:
Agree